diff --git a/core/lib/Drupal/Core/Database/Connection.php b/core/lib/Drupal/Core/Database/Connection.php
index b3cd1c5b0d..3850a39100 100644
--- a/core/lib/Drupal/Core/Database/Connection.php
+++ b/core/lib/Drupal/Core/Database/Connection.php
@@ -87,9 +87,9 @@
   protected $transactionalDDLSupport = FALSE;
 
   /**
-   * The actual PDO connection.
+   * The actual client connection.
    *
-   * @var \PDO
+   * @var object
    */
   protected $connection;
 
@@ -803,7 +803,7 @@ public function query($query, array $args = [], $options = []) {
 
         case Database::RETURN_INSERT_ID:
           $sequence_name = $options['sequence_name'] ?? NULL;
-          return $this->connection->lastInsertId($sequence_name);
+          return $this->lastInsertId($sequence_name);
 
         case Database::RETURN_NULL:
           return NULL;
diff --git a/core/modules/mysql/src/Driver/Database/mysql/Connection.php b/core/modules/mysql/src/Driver/Database/mysql/Connection.php
index e15575bcb0..49b4075398 100644
--- a/core/modules/mysql/src/Driver/Database/mysql/Connection.php
+++ b/core/modules/mysql/src/Driver/Database/mysql/Connection.php
@@ -271,7 +271,7 @@ protected function getMariaDbVersionMatch(): ?string {
    */
   protected function getServerVersion(): string {
     if (!$this->serverVersion) {
-      $this->serverVersion = $this->connection->query('SELECT VERSION()')->fetchColumn();
+      $this->serverVersion = $this->query('SELECT VERSION()')->fetchField();
     }
     return $this->serverVersion;
   }
diff --git a/core/tests/Drupal/KernelTests/Core/Database/DatabaseExceptionWrapperTest.php b/core/tests/Drupal/KernelTests/Core/Database/DatabaseExceptionWrapperTest.php
index 0fd2770fe5..f72e7e527a 100644
--- a/core/tests/Drupal/KernelTests/Core/Database/DatabaseExceptionWrapperTest.php
+++ b/core/tests/Drupal/KernelTests/Core/Database/DatabaseExceptionWrapperTest.php
@@ -20,8 +20,16 @@ class DatabaseExceptionWrapperTest extends KernelTestBase {
    * class, which defer the statement check to the moment of the execution.
    */
   public function testPrepareStatementFailOnExecution() {
+    $connection = Database::getConnection();
+    $connection_reflection_class = new \ReflectionClass($connection);
+    $client_connection_property = $connection_reflection_class->getProperty('connection');
+    $client_connection_property->setAccessible(TRUE);
+    $client_connection = $client_connection_property->getValue($connection);
+    if (!$client_connection instanceof \PDO) {
+      $this->markTestSkipped("This tests can only run for drivers wrapping \\PDO connections.");
+    }
     $this->expectException(\PDOException::class);
-    $stmt = Database::getConnection()->prepareStatement('bananas', []);
+    $stmt = $connection->prepareStatement('bananas', []);
     $stmt->execute();
   }
 
diff --git a/core/tests/Drupal/Tests/Core/Database/Driver/mysql/ConnectionTest.php b/core/tests/Drupal/Tests/Core/Database/Driver/mysql/ConnectionTest.php
index e8d6370225..53d09f0596 100644
--- a/core/tests/Drupal/Tests/Core/Database/Driver/mysql/ConnectionTest.php
+++ b/core/tests/Drupal/Tests/Core/Database/Driver/mysql/ConnectionTest.php
@@ -2,8 +2,10 @@
 
 namespace Drupal\Tests\Core\Database\Driver\mysql;
 
+use Drupal\Core\Database\StatementWrapper;
 use Drupal\mysql\Driver\Database\mysql\Connection;
 use Drupal\Tests\UnitTestCase;
+use Prophecy\Argument;
 
 /**
  * Tests MySQL database connections.
@@ -41,6 +43,21 @@ protected function setUp(): void {
    * @return \Drupal\mysql\Driver\Database\mysql\Connection
    */
   private function createConnection(): Connection {
+    $this->pdoStatement
+      ->setFetchMode(Argument::any())
+      ->shouldBeCalled()
+      ->willReturn(TRUE);
+
+    $this->pdoStatement
+      ->execute(Argument::any())
+      ->shouldBeCalled()
+      ->willReturn(TRUE);
+
+    $this->pdoConnection
+      ->prepare('SELECT VERSION()', Argument::any())
+      ->shouldBeCalled()
+      ->willReturn($this->pdoStatement->reveal());
+
     /** @var \PDO $pdo_connection */
     $pdo_connection = $this->pdoConnection->reveal();
 
@@ -60,15 +77,10 @@ public function __construct(\PDO $connection) {
    */
   public function testVersionAndIsMariaDb(bool $expected_is_mariadb, string $server_version, string $expected_version): void {
     $this->pdoStatement
-      ->fetchColumn()
+      ->fetchColumn(Argument::any())
       ->shouldBeCalled()
       ->willReturn($server_version);
 
-    $this->pdoConnection
-      ->query('SELECT VERSION()')
-      ->shouldBeCalled()
-      ->willReturn($this->pdoStatement->reveal());
-
     $connection = $this->createConnection();
 
     $is_mariadb = $connection->isMariaDb();
