diff --git a/core/modules/config/tests/src/Functional/ConfigInstallProfileUnmetDependenciesTest.php b/core/modules/config/tests/src/Functional/ConfigInstallProfileUnmetDependenciesTest.php
index c459f15250..12a445e7f8 100644
--- a/core/modules/config/tests/src/Functional/ConfigInstallProfileUnmetDependenciesTest.php
+++ b/core/modules/config/tests/src/Functional/ConfigInstallProfileUnmetDependenciesTest.php
@@ -49,12 +49,12 @@ protected function prepareEnvironment() {
   protected function setUp(): void {
     // During set up an UnmetDependenciesException should be thrown, which will
     // be re-thrown by TestHttpClientMiddleware as a standard Exception.
-    try {
+//    try {
       parent::setUp();
-    }
+/*    }
     catch (\Exception $exception) {
       $this->expectedException = $exception;
-    }
+    }*/
   }
 
   /**
diff --git a/core/tests/Drupal/Tests/UiHelperTrait.php b/core/tests/Drupal/Tests/UiHelperTrait.php
index be7a0c7c1a..cf6570ca07 100644
--- a/core/tests/Drupal/Tests/UiHelperTrait.php
+++ b/core/tests/Drupal/Tests/UiHelperTrait.php
@@ -80,6 +80,7 @@ protected function submitForm(array $edit, $submit, $form_html_id = NULL) {
     }
 
     // Edit the form values.
+dump($edit);
     foreach ($edit as $name => $value) {
       $field = $assert_session->fieldExists($name, $form);
 
