diff --git a/core/modules/page_cache/tests/src/Functional/PageCacheTagsIntegrationTest.php b/core/modules/page_cache/tests/src/Functional/PageCacheTagsIntegrationTest.php
index 7c127fe41b..3a8689cdcf 100644
--- a/core/modules/page_cache/tests/src/Functional/PageCacheTagsIntegrationTest.php
+++ b/core/modules/page_cache/tests/src/Functional/PageCacheTagsIntegrationTest.php
@@ -84,6 +84,7 @@ public function testPageCacheTags() {
     ];
 
     // Full node page 1.
+global $xxx; $xxx=TRUE;
     $this->assertPageCacheContextsAndTags($node_1->toUrl(), $cache_contexts, [
       'http_response',
       'rendered',
diff --git a/core/modules/system/tests/src/Functional/Cache/AssertPageCacheContextsAndTagsTrait.php b/core/modules/system/tests/src/Functional/Cache/AssertPageCacheContextsAndTagsTrait.php
index fefd3a462b..2946306974 100644
--- a/core/modules/system/tests/src/Functional/Cache/AssertPageCacheContextsAndTagsTrait.php
+++ b/core/modules/system/tests/src/Functional/Cache/AssertPageCacheContextsAndTagsTrait.php
@@ -69,6 +69,7 @@ protected function assertNoCacheContext($not_expected_cache_context) {
    */
   protected function assertPageCacheContextsAndTags(Url $url, array $expected_contexts, array $expected_tags) {
     $absolute_url = $url->setAbsolute()->toString();
+global $xxx; if ($xxx) dump([$absolute_url, $url]);
     sort($expected_contexts);
     sort($expected_tags);
 
