diff --git a/core/lib/Drupal/Core/Database/StatementWrapper.php b/core/lib/Drupal/Core/Database/StatementWrapper.php
index 32e0bf5825..fbdd13256f 100644
--- a/core/lib/Drupal/Core/Database/StatementWrapper.php
+++ b/core/lib/Drupal/Core/Database/StatementWrapper.php
@@ -366,6 +366,7 @@ protected function markResultsetFetchingComplete(): void {
    * @internal This method should not be called directly.
    */
   public function current(): mixed {
+dump([__METHOD__, $this->resultsetRow]);
     return $this->resultsetRow;
   }
 
@@ -375,6 +376,7 @@ public function current(): mixed {
    * @internal This method should not be called directly.
    */
   public function key(): mixed {
+dump([__METHOD__, $this->resultsetKey]);
     return $this->resultsetKey;
   }
 
@@ -384,6 +386,7 @@ public function key(): mixed {
    * @internal This method should not be called directly.
    */
   public function rewind(): void {
+dump([__METHOD__]);
     // Nothing to do: our DatabaseStatement can't be rewound.
   }
 
@@ -393,6 +396,7 @@ public function rewind(): void {
    * @internal This method should not be called directly.
    */
   public function next(): void {
+dump([__METHOD__]);
     $this->fetch();
   }
 
@@ -402,6 +406,7 @@ public function next(): void {
    * @internal This method should not be called directly.
    */
   public function valid(): bool {
+dump([__METHOD__, $this->isResultsetIterable, $this->resultsetKey]);
     if ($this->isResultsetIterable && $this->resultsetKey === -1) {
       $this->fetch();
     }
